honeycomb-core/src/geometry/dim3/tests.rs

```diff
@@ -0,0 +1,220 @@
+// ------ IMPORTS
+
+use super::super::{CoordsFloat, Vector3, Vertex3};
+
+// ------ CONTENT
+
+// --- common
+// scalar equality function
+// ref: https://floating-point-gui.de/errors/comparison/
+fn almost_equal<T: CoordsFloat>(a: T, b: T) -> bool {
+    let abs_diff = (a - b).abs();
+    let abs_sum = a.abs() + b.abs();
+    if a == b {
+        // early return
+        true
+    } else if a.is_zero() | b.is_zero() | (abs_sum < T::min_positive_value()) {
+        // close to zero
```

Comment:
Is `T::min_positive_value()` the smallest positive *normal* value?

I do not quite follow this path. Why is using a non-zero *denormal* value a problem?